src/log-store/src/kafka.rs

```diff
@@ -69,6 +70,10 @@ impl Entry for EntryImpl {
     fn namespace(&self) -> Self::Namespace {
         self.ns.clone()
     }
+
+    fn estimated_size(&self) -> usize {
+        size_of::<Self>() + self.data.capacity() * size_of::<u8>() + self.ns.topic.capacity()
+    }
```

Comment:
Is this method meant to calculate the runtime memory footprint of entries other than the actual serialized data size? bcz I saw it's used in metric `METRIC_KAFKA_APPEND_BATCH_BYTES_TOTAL`.

Reply:
Yes, it calculates the runtime memory footprint. This metric is meant to calculate how many bytes in total are injected into the log store.